apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -336,6 +337,13 @@ type EnvoyListenerConfig struct {
 	// +optional
 	DisableMergeSlashes *bool `json:"disableMergeSlashes,omitempty"`
 
+	// DisableServerHeaderTransformation signifies Envoy will not modify the Server header.
+	// It has 3 possible values: OVERWRITE, PASS_THROUGH, APPEND_IF_ABSENT
+	//
+	// Contour's default is OVERWRITE.
+	// +optional
+	DisableServerHeaderTransformation *http.HttpConnectionManager_ServerHeaderTransformation `json:"disableServerHeaderTransformation,omitempty"`
```

Comment:
Let's change the name of this field to `ServerHeaderTransformation` since it's now an enum rather than a bool (including all godoc, JSON tags, etc)